ha/setup/ha_setup.py

```diff
@@ -791,6 +790,69 @@ def process(self):
         """
         pass # TBD
 
+
+class PreUpgradeCmd(Cmd):
+    """
+    Pre-Upgrade Cmd
+    """
+    name = "pre_upgrade"
+
+    def __init__(self, args):
+        """
+        Init method.
+        """
+        super().__init__(args)
+
+    def process(self):
+        """
+        Process command.
+
+        NOTE: Consul is not expected to be upgraded, so no need to backup
+        consul config
+        Configuration backup is performed for every node while pacemaker setup
+        only on one node (cluster level).
+        """
+        try:
+            if os.environ['PRVSNR_MINI_LEVEL'] == 'cluster':
+                Log.info("Performing pre disruptive upgrade routines on cluster \
+                         level")
+                cluster_standby_mode()
+                delete_resources()
+            else:
+                Log.info("Performing pre disruptive upgrade routines on node \
+                         level")
+                backup_configuration()
+        except Exception as err:
+            raise SetupError("Pre-upgrade routines failed") from err
+
+
+class PostUpgradeCmd(Cmd):
+    """
+    Post-Upgrade Cmd
+    """
+    name = "post_upgrade"
+
+    def __init__(self, args):
+        """
+        Init method.
+        """
+        super().__init__(args)
+        machine_id = self.get_machine_id()
+        self.s3_instance = UpgradeCmd.get_s3_instance(machine_id)
+
+    def process(self):
+        """
+        Process command.
+        """
+        try:
+            if os.environ['PRVSNR_MINI_LEVEL'] == 'cluster':
+                Log.info("Performing post disruptive upgrade routines on cluster \
+                         level")
+                perform_post_upgrade(self.s3_instance)
```

Comment:
as I see [here](https://github.com/Seagate/cortx-ha/blob/9fc5ba6dc1a3ceb78e84b46ee8daaf9a6afa2511/ha/setup/post_disruptive_upgrade.py#L163) unstandby would be called as part of post-upgrade. Generally it's not accurate since other components may rely on the documented sequence where unstandby/start is called after post-upgrade calls are done for all the components.

Reply:
Updated post upgrade script. Thanks.